utils/base64.cc

```diff
@@ -128,7 +128,7 @@ bool base64_begins_with(std::string_view base, std::string_view operand) {
         return false;
     }
     // Decode and compare last 4 bytes of base64-encoded strings
-    const std::string base_remainder = base64_decode_string(base.substr(operand.size() - 4, operand.size()));
+    const std::string base_remainder = base64_decode_string(base.substr(operand.size() - 4, 4));  // BUG FIX 8: Fixed substr length parameter
```

Comment:
I think this comment is correct, it's a mistake - whoever wrote this (Piotr Sarna, 5 years ago) probably mistook the second parameter to be the end of the substring - not the length.

However, this typo will never cause a real bug in practice:

1. From previous logic know that operand_size() is a multiple of 4 and not empty, so 4 and higher.
2. So substr(operand.size() - 4, operand.size() will always return the last 4 characters, because operand_size() > 4 and it's not an error to use substr with length longer than what we have - we'll only get those 4 characters.